server/src/services/runs.service.ts

```diff
@@ -0,0 +1,125 @@
+import { Request, Response } from 'express';
+import { validationResult } from 'express-validator';
+import { AppDataSource } from '../../orm.config';
+import { Run } from '../entities/Run';
+
+export class RunsService {
+  async createRun(req: Request, res: Response): Promise<Response> {
+    try {
+      const errors = validationResult(req);
```

Comment:
let's take out this check, dry is violated. To do this you need to throw from the new reusable function, catch the error somewhere and send response with according error message and status code